net/Socket.cpp

```diff
@@ -92,22 +92,21 @@ std::string Socket::toString(Type t)
 
 int Socket::createSocket([[maybe_unused]] Socket::Type type)
 {
-    if constexpr (!Util::isMobileApp())
+#if !MOBILEAPP
```

Comment:
The idea to deal with the compiler seeing unsupported functions is to stub them. We are trying to avoid compile-time directives because the code behind them can rot (at least that's my favorite reason, but there are others, including making code harder to read).

Reply:
Yes, that is how I will do it in the future. But this PR has commits that are almost a month old by now... before I came to the conclusion that what you describe indeed is the best way. Ifdefs will be removed in later commits.